table/queries/derived_satellite_scans.sql

```diff
@@ -0,0 +1,205 @@
+    WHEN rcode = 2 THEN "❗️dns/rcode:ServFail"
+    WHEN rcode = 3 THEN "❗️dns/rcode:NXDomain"
+    WHEN rcode = 4 THEN "❗️dns/rcode:NotImp"
+    WHEN rcode = 5 THEN "❗️dns/rcode:Refused"
+    WHEN rcode = 6 THEN "❗️dns/rcode:YXDomain"
+    WHEN rcode = 7 THEN "❗️dns/rcode:YXRRSet"
+    WHEN rcode = 8 THEN "❗️dns/rcode:NXRRSet"
+    WHEN rcode = 9 THEN "❗️dns/rcode:NotAuth"
+    WHEN rcode = 10 THEN "❗️dns/rcode:NotZone"
+    WHEN rcode = 11 THEN "❗️dns/rcode:DSOTYPENI"
+    WHEN rcode = 12 THEN "❗️dns/rcode:Unassigned"
+    WHEN rcode = 13 THEN "❗️dns/rcode:Unassigned"
+    WHEN rcode = 14 THEN "❗️dns/rcode:Unassigned"
+    WHEN rcode = 15 THEN "❗️dns/rcode:Unassigned"
+    WHEN rcode = 16 THEN "❗️dns/rcode:BadVers"
+    WHEN rcode = 17 THEN "❗️dns/rcode:BadSig"
+    WHEN rcode = 18 THEN "❗️dns/rcode:BadKey"
+    WHEN rcode = 19 THEN "❗️dns/rcode:BadTime"
+    WHEN rcode = 20 THEN "❗️dns/rcode:BadMode"
+    WHEN rcode = 21 THEN "❗️dns/rcode:BadAlg"
+    WHEN rcode = 22 THEN "❗️dns/rcode:BadTrunc"
+    WHEN rcode = 23 THEN "❗️dns/rcode:BadCookie"
+    ELSE CONCAT("❗️dns/unknown_rcode:", rcode)
+  END
+);
+
+CREATE TEMP FUNCTION ClassifySatelliteError(error STRING) AS (
+  CASE
+    # Satellite v1
+    WHEN REGEXP_CONTAINS(error, '"Err": {}') THEN "❗️read/udp.timeout"
+    WHEN REGEXP_CONTAINS(error, '"Err": 90') THEN "❗️read/dns.msgsize"
+    WHEN REGEXP_CONTAINS(error, '"Err": 111') THEN "❗️read/udp.refused"
+    WHEN REGEXP_CONTAINS(error, '"Err": 113') THEN "❗️read/ip.host_no_route"
+    WHEN REGEXP_CONTAINS(error, '"Err": 24') THEN "❗️setup/system_failure" # Too many open files
+    WHEN error = "{}" THEN "❗️dns/unknown" # TODO figure out origin
+    WHEN error = "no_answer" THEN "❗️dns/answer:no_answer"
+    #Satellite v2
+    WHEN ENDS_WITH(error, "i/o timeout") THEN "❗️read/udp.timeout"
+    WHEN ENDS_WITH(error, "message too long") THEN "❗️read/dns.msgsize"
+    WHEN ENDS_WITH(error, "connection refused") THEN "❗️read/udp.refused"
+    WHEN ENDS_WITH(error, "no route to host") THEN "❗️read/ip.host_no_route"
+    WHEN ENDS_WITH(error, "short read") THEN "❗️read/dns.msgsize"
+    WHEN ENDS_WITH(error, "read: protocol error") THEN "❗️read/protocol_error"
+    WHEN ENDS_WITH(error, "socket: too many open files") THEN "❗️setup/system_failure"
+    ELSE CONCAT("❗️dns/unknown_error:", error)
+  END
+);
+
+CREATE TEMP FUNCTION InvalidIpType(ip STRING) AS (
+  CASE
+    WHEN STARTS_WITH(ip, "0.") THEN "❗️ip_invalid:zero"
+    WHEN STARTS_WITH(ip, "127.") THEN "❗️ip_invalid:local_host"
+    WHEN STARTS_WITH(ip, "10.") THEN "❗️ip_invalid:local_net"
+    WHEN NET.IP_TO_STRING(NET.IP_TRUNC(NET.SAFE_IP_FROM_STRING(ip), 12)) = "172.16.0.0"  THEN "❗️ip_invalid:local_net"
+    WHEN STARTS_WITH(ip, "192.168.") THEN "❗️ip_invalid:local_net"
+    ELSE NULL
+  END
+);
+
+CREATE TEMP FUNCTION ClassifySatelliteErrorNegRCode(error STRING) AS (
+  CASE
+    WHEN (error IS NULL OR error = "" OR error = "null")  THEN "❗️read/udp.timeout"
```

Comment:
moved the `rcode=-1` statement, so this code is no longer dead.

I don't think those cases overlapping is common, and it would only happen in the v1 data, but that's the right way to handle it.